src/csharp/Microsoft.Spark.E2ETest/UdfTests/UdfComplexTypesTests.cs

```diff
@@ -0,0 +1,191 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Spark.Sql;
+using Microsoft.Spark.Sql.Types;
+using Xunit;
+using static Microsoft.Spark.Sql.Functions;
+
+namespace Microsoft.Spark.E2ETest.UdfTests
+{
+    [Collection("Spark E2E Tests")]
+    public class UdfComplexTypesTests
+    {
+        private readonly SparkSession _spark;
+        private readonly DataFrame _df;
+
+        public UdfComplexTypesTests(SparkFixture fixture)
+        {
+            _spark = fixture.Spark;
+            _df = _spark
+                .Read()
+                .Json(Path.Combine($"{TestEnvironment.ResourceDirectory}people.json"));
+        }
+
+        /// <summary>
+        /// UDF that takes in Array type.
+        /// </summary>
+        [Fact]
+        public void TestUdfWithArrayType()
+        {
+            // UDF with ArrayType throws a following exception:
+            // [] [] [Error] [TaskRunner] [0] ProcessStream() failed with exception: System.InvalidCastException: Unable to cast object of type 'System.Collections.ArrayList' to type 'System.Int32[]'.
+            //  at Microsoft.Spark.Sql.PicklingUdfWrapper`2.Execute(Int32 splitIndex, Object[] input, Int32[] argOffsets) in Microsoft.Spark\Sql\PicklingUdfWrapper.cs:line 44
+            //  at Microsoft.Spark.Worker.Command.PicklingSqlCommandExecutor.SingleCommandRunner.Run(Int32 splitId, Object input) in Microsoft.Spark.Worker\Command\SqlCommandExecutor.cs:line 239
+            //  at Microsoft.Spark.Worker.Command.PicklingSqlCommandExecutor.ExecuteCore(Stream inputStream, Stream outputStream, SqlCommand[] commands) in Microsoft.Spark.Worker\Command\SqlCommandExecutor.cs:line 139
+            Func<Column, Column> udf = Udf<int[], string>(array => string.Join(',', array));
+            Assert.Throws<Exception>(() => _df.Select(udf(_df["ids"])).Show());
+
+            // Currently, there is a workaround to support ArrayType using ArrayList. See the example below.
+            Func<Column, Column> workingUdf = Udf<ArrayList, string>(array => string.Join(',', array.ToArray()));
+
+            Row[] rows = _df.Select(workingUdf(_df["ids"])).Collect().ToArray();
+            Assert.Equal(3, rows.Length);
+            
+            var expected = new[] { "1", "3,5", "2,4" };
+            for (int i = 0; i < rows.Length; ++i)

```

Comment:
Can you replace the below with something like `Assert.Equal(expected, rows)` ?

Reply:
I have made some changes to `Assert.Equal()`. Please let me know if this makes sense. Thanks!